src/Microsoft.ML.AutoML/Utils/SplitUtil.cs

```diff
@@ -47,9 +47,12 @@ internal static class SplitUtil
         /// Split the data into a single train/test split.
         /// </summary>
         public static (IDataView trainData, IDataView validationData) TrainValidateSplit(MLContext context, IDataView trainData,
-            string samplingKeyColumn)
+            string samplingKeyColumn, bool shuffle = false)
```

Comment:
Perhaps conceptually 'shuffle' belongs outside of the train/valid split API ?
If desired (and only desired for images right now), you could shuffle the results of the this API
For instance, I don't think it's part of the ML.NET train/valid split API

Reply:
We need to shuffle otherwise you can train on set of images and test on set of images that you did not train on, images need to be shuffled before they are fed into the trainer. Think of a dataset that has images of cats followed by images of dogs, now whether you do CV or TrainTest, you will end up training on cats and testing on dogs and get a very poor accuracy.

---
In reply to: [339248471](https://github.com/dotnet/machinelearning/pull/4384#discussion_r339248471) [](ancestors = 339248471)

Reply:
See comment above on the non-need of shuffling for TrainValidate/CV splits: https://github.com/dotnet/machinelearning/pull/4384/files#r339313039